nixos/modules/services/mail/goeland.nix

```diff
@@ -0,0 +1,59 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.goeland;
+  tomlFormat = pkgs.formats.toml { };
+in
+{
+  options.services.goeland = {
+    enable = mkEnableOption (mdDoc "goeland");
+
+    settings = mkOption {
+      description = mdDoc ''
+        Configuration of goeland.
+        See the [example config file](https://github.com/slurdge/goeland/blob/master/cmd/asset/config.default.toml) for the available options.
+      '';
+      default = { };
+      type = types.submodule {
+        freeformType = tomlFormat.type;
+      };
+    };
+    schedule = mkOption {
+      type = types.str;
+      default = "12h";
+      example = "Mon, 00:00:00";
+      description = mdDoc "How often to run goeland, in systemd time format";
+    };
+    databaseDirectory = mkOption {
+      type = types.path;
+      default = "/var/lib/goeland";
```

Comment:
It shouldn't break the service, it will just create a non-used /var/lib/goeland which won't be the directory used by goeland to store the database.
Here's my reasoning here: https://github.com/NixOS/nixpkgs/pull/206099#discussion_r1063493234